src/NuGet.Core/NuGet.Commands/PackagesLockFileBuilder.cs

```diff
@@ -89,5 +94,56 @@ public PackagesLockFile CreateNuGetLockFile(LockFile assetsFile)
             return lockFile;
         }
 
+        public async Task<PackagesLockFile> CreateNuGetLockFileAsync(IEnumerable<PackageReference> installedPackages, Func<PackageIdentity, string> getNupkgFilePath, CancellationToken token)
+        {
+            Debug.Assert(installedPackages != null);
+            Debug.Assert(getNupkgFilePath != null);
+
+            var lockFile = new PackagesLockFile();
+
+            foreach (var targetFramework in installedPackages.GroupBy(p => p.TargetFramework).OrderBy(g => g.Key))
+            {
+                var target = new PackagesLockFileTarget
+                {
+                    TargetFramework = targetFramework.Key,
+                    Dependencies = new List<LockFileDependency>()
+                };
+
+                foreach (var package in targetFramework.OrderBy(p => p.PackageIdentity))
+                {
+                    string hash;
+
+                    var file = getNupkgFilePath(package.PackageIdentity);
+                    if (file == null)
+                    {
+                        throw new Exception("couldn't find package");
```

Comment:
Multiple concerns here:
Should it really throw at this point?
Should it really throw a generic Exception? without any relevant details? Resource string?

Reply:
oops, I intended this to be temporary and to "do it properly" before commiting, but I forgot :(  As part of fixing other issues, my code no longer throws any exceptions.